internal/provider/resource_vdb.go

```diff
@@ -567,6 +568,14 @@ func resourceVdb() *schema.Resource {
 					},
 				},
 			},
+			"appdata_source_params": {
+				Type:     schema.TypeString,
```

Comment:
This String type can be confusing to user? Have we considered using https://developer.hashicorp.com/terraform/plugin/sdkv2/schemas/schema-types#typemap ? and converting it to JSON? 

TypeMap can take any key,value so it is still dynamic and not tied to specific payload fields

Reply:
We have explored the possibility of typemap but the challenge here is:

1. in typemap we got to define the types of the child payload thats coming in. And whatever we defined there should hold true for all the child keys. For eg: if we introduce something like 
`Type:     schema.TypeMap,
  Elem: &schema.Schema{
    Type: schema.TypeString,
  },`
then the key values should be all strings. which is not the case here.

2. Now we can also define specific types in the child elements of the json, but that is also not possible for us because we can have any new parameter in the json provided. This is because every plugin developer can write the parameters required by their plugin in their own way. For instance, if postgre has a key called configsettings, Hana can have the same key called configsettingsstg. There is no standardization between different plugins. 

Hence, we opted for this approach which will hold true for all types of plugins.